src/shared/credentials/accountId.ts

```diff
@@ -26,3 +27,27 @@ export async function getAccountId(credentials: AWS.Credentials, region: string)
         return undefined
     }
 }
+
+/**
+ * Temporary workaround for missing credentials bug.
+ * TODO: Remove this function when migrated over to V3 of the SDK
+ */
+export async function getAccountIdHack(credentials: AWS.Credentials, region: string): Promise<string | undefined> {
```

Comment:
What's the likelihood/risk of concurrently running code getting an unexpected side effect caused by this switcheroo?

Reply:
I've thought about this, and from what I've looked into the risk appears to be fairly low. The environment variable is ignored if we pass the region through the constructor. I'm not sure why the bug even exists, but I guess the SDK attempts to read the region from the credentials file regardless of the provided region.